### DISCUSSION TITLE
unable to access Dovecot

### DISCUSSION THREAD

**Discussion by oschwand** [Mailu/Mailu#3140] [User support] [answered]
## Environment & Version

    $ docker-compose version
    docker-compose version 1.25.0, build unknown

Mailu version: 2.0

I can provide my docker-compose.yml and my mailu.env files but github seems to allow a very short list of file formats...

## Description

It's not possible to log to dovecot (and thus to the webmail). According to the logs it comes from /var/run/dovecot/login but it seems to be a more general issues.

A work around is to run the following inside the imap container after starting the container:

```
chgrp -R mail /var/run/dovecot
chmod g+rwX -R /var/run/dovecot
```

I may put it inside my docker-compose.yml file but I would prefer a more convincing solution. It seems to be related to https://github.com/docker-mailserver/docker-mailserver/issues/1069.

## Replication Steps

Just launch the stack with `docker-compose up`

## Observed behaviour

Login failure.

In addition to the login failure, the webmail fails completely with infinite redirect. I think it's probably a different issue with the sso when the login fails in some weird ways. From time to time the webmail is displayed but obviously with an error message about the storage.

## Expected behaviour

Just a login.

## Logs

A relevant fragment of logs is

```
imap_1        | Jan 21 20:24:18 imap-login: Error: auth-client: conn unix:login: connect(login) in directory / failed: Permission denied (euid=8(<getpwuid() error>) egid=12(<getgrgid() error>) missing +w perm: //login, dir owned by 0:12 mode=0750)
imap_1        | Jan 21 20:24:18 imap-login: Error: auth-client: conn unix:login: connect(login) in directory / failed: Permission denied (euid=8(<getpwuid() error>) egid=12(<getgrgid() error>) missing +w perm: //login, dir owned by 0:12 mode=0750)
imap_1        | Jan 21 20:24:28 imap-login: Warning: Auth process not responding, delayed sending initial response (greeting): user=<>, rip=192.168.0.2, lip=192.168.0.3, session=<UnFzqXkPSIjAqAAC>
imap_1        | Jan 21 20:24:28 imap-login: Info: Disconnected: Connection closed (disconnected before auth was ready, waited 10 secs): user=<>, rip=192.168.0.2, lip=192.168.0.3, session=<UnFzqXkPSIjAqAAC>
```

**Comment by oschwand** [comment]
After looking at /start.py inside the imap container, it seems a solution would be to add a line about /var/run/dovecot in the chown section at the end.

EDIT:

No it's not working, dovecot is "fixing" the permissions in some way:

```
Warning: Corrected permissions for login directory /run/dovecot/login
```

**Reply by nextgens** [reply]
The ticket you have linked suggests that it is a permission/ACL issue on the host; start by looking there since it works for others.

**Reply by oschwand** [reply, marked as answer]
Ok, you're right. For some reasons there was a default acl on my /var/lib/docker, no idea why...

I solved the problem with `setfacl` -k /var/lib/docker/` and removed the images to ensure a clean state.

Thx
